staging/src/k8s.io/client-go/kubernetes/typed/apps/v1/apps_client.go

```diff
@@ -163,3 +173,30 @@ func (c *AppsV1Client) RESTClients() []rest.Interface {
 
 	return c.restClients
 }
+
+// run watch api server instance updates and recreate connections to new set of api servers
+func (c *AppsV1Client) run() {
+	go func(c *AppsV1Client) {
+		member := c.configs.WatchUpdate()
+		watcherForUpdateComplete := apiserverupdate.GetClientSetsWatcher()
+		watcherForUpdateComplete.AddWatcher()
+
+		for {
+			select {
```

Comment:
If there is only one channel in the select statement,  a simple loop will be sufficient.

Reply:
Please check https://golang.org/src/runtime/select.go#L208
In select statement implementation, it will lock all the channels first.

Reply:
updated